src/cpp/simplehttpserver.cpp

```diff
@@ -553,7 +553,10 @@ SimpleHttpRequest *SimpleHttpServer::takeNext()
 	if(!d->pending.isEmpty())
 	{
 		SimpleHttpRequest *req = d->pending.takeFirst();
-		req->disconnect(d);
+		auto conn = d->finishedConnections.find(req);
+		if (conn != d->finishedConnections.end())
+			d->finishedConnections.erase(conn);
```

Comment:
You may even be able to write this more simply as: `d->finishedConnections.erase(req)`.